### PR TITLE
Remove the Subscriptions feature

### PR DESCRIPTION
**Pull request by jtherrmann**
## Overview

I confirmed that each of the following commands shows no output:

```bash
# finds all tracked filenames containing "subscription" (case-insensitive)
git ls-tree -r remove-subscriptions --name-only | grep -i subscription

# finds all occurrences of "subscription" (case-insensitive) in tracked files other than the Changelog
git grep -i subscription | grep -v CHANGELOG
```

We expect these resources to get deleted:
- subscription manager and worker lambdas
- subscriptions dynamodb table
- subscription_id parameter for get jobs api endpoint
- subscriptions api endpoints
- subscriptions-related api schemas

## Before merging to test

- [x] Changelog entry.

Verify changes in a sandbox deployment:

- [x] Create a new sandbox deployment (branch off of `develop`).
- [x] Set user quota for myself.
- [x] Create some subscriptions in the sandbox and let them generate some products.
- [x] Merge `remove-subscriptions` into `subscriptions-removal-sandbox` and verify that all expected resources get deleted
- [x] Why does Swagger UI show INVALID for API schema? (see bottom-right corner of page)
- [x] Manually remove the `subscription_id` field from the Jobs table.

## After merging to test

- [ ] Verify that all expected resources get deleted
- [ ] Verify that Swagger UI shows VALID for API schema (see bottom-right corner of page)
- [ ] Manually remove the `subscription_id` field from the Jobs table.

### CODE REVIEW

File: apps/api/src/hyp3_api/api-spec/job_parameters.yml.j2
Comment (asjohnston-asf) [inline apps/api/src/hyp3_api/api-spec/job_parameters.yml.j2:1]: We can remove the `job_specification` schema. I suspect all references have already been removed but haven't double-checked yet.

File: apps/api/src/hyp3_api/routes.py
Comment (asjohnston-asf) [inline apps/api/src/hyp3_api/routes.py:1]: I suspect the `WKTValidator` class is no longer needed.

File: apps/api/src/hyp3_api/api-spec/job_parameters.yml.j2
Comment (jtherrmann) [reply apps/api/src/hyp3_api/api-spec/job_parameters.yml.j2:1]: Good catch, I removed the schema and confirmed that all references have been removed.

File: apps/api/src/hyp3_api/routes.py
Comment (jtherrmann) [reply apps/api/src/hyp3_api/routes.py:1]: Good catch, I removed it. I should probably find a way to do a comprehensive sweep for any unused functions and classes.

Comment (jtherrmann) [reply apps/api/src/hyp3_api/routes.py:1]: I ran https://pypi.org/project/vulture/ on `apps/` and `lib/`, on both `develop` and `remove-subscriptions`, to confirm that no more unused code has appeared on the `remove-subscriptions` branch.

**Review by asjohnston-asf — APPROVED** [review]

**Pull request merged**